File: benchmark/fortran/task5_class.f90
module point_mod
    implicit none
    type :: Point
        real :: x, y
    contains
        procedure :: distance_to
    end type Point
contains
    function distance_to(self, other) result(d)
        class(Point), intent(in) :: self, other
        real :: d, dx, dy
        dx = self%x - other%x
        dy = self%y - other%y
        d = sqrt(dx*dx + dy*dy)
    end function distance_to
end module point_mod

File: benchmark/matlab/task5_class.m
classdef Point
    properties
        x
        y
    end
    methods
        function obj = Point(x, y)
            obj.x = x;
            obj.y = y;
        end
        function d = distance_to(obj, other)
            dx = obj.x - other.x;
            dy = obj.y - other.y;
            d = sqrt(dx^2 + dy^2);
        end
    end
end
